src/main/java/bio/terra/service/filedata/flight/ingest/IngestFilePrimaryDataLocationStep.java

```diff
@@ -32,8 +34,13 @@ public StepResult doStep(FlightContext context) {
         if (loadComplete == null || !loadComplete) {
             FlightMap inputParameters = context.getInputParameters();
             FileLoadModel fileLoadModel = inputParameters.get(JobMapKeys.REQUEST.getKeyName(), FileLoadModel.class);
-            GoogleBucketResource bucketForFile = locationService.getOrCreateBucketForFile(fileLoadModel.getProfileId());
-            workingMap.put(FileMapKeys.BUCKET_INFO, bucketForFile);
+            try {
+                GoogleBucketResource bucketForFile =
+                    locationService.getOrCreateBucketForFile(fileLoadModel.getProfileId(), context.getFlightId());
+                workingMap.put(FileMapKeys.BUCKET_INFO, bucketForFile);
+            } catch (BucketLockException blEx) {
```

Comment:
We define the retry rule in the flight, so it makes more sense to me to specify a particular error as being retry-able in the steps, rather than pushing it down into the service class. Or maybe from a different side, I didn't want to tie this particular exception to a retry, in case we decide it's not retry-able in a different context/flight. I agree the behavior here would be the same either way.

Reply:
I agree, if you want to use the same exception in retry/no-retry contexts, what you did is the way to go.